packages/phone-number-privacy/signer/src/database/wrappers/request.ts

```diff
@@ -0,0 +1,47 @@
+import { ErrorMessage } from '@celo/phone-number-privacy-common'
+import { DB_TIMEOUT } from '../../common/constants'
+import logger from '../../common/logger'
+import { GetBlindedMessageForSaltRequest } from '../../salt-generation/get-salt'
+import { getDatabase } from '../database'
+import { Request, REQUESTS_COLUMNS, REQUESTS_TABLE } from '../models/request'
+
+function requests() {
+  return getDatabase()<Request>(REQUESTS_TABLE)
+}
+
+export async function getRequestExists(request: GetBlindedMessageForSaltRequest): Promise<boolean> {
+  if (!request.timestamp) {
+    return false // TODO(Alec) make timestamps required
+  }
+  logger.debug('Checking if request exists')
+  try {
+    const existingRequest = await requests()
+      .where(REQUESTS_COLUMNS.timestamp, new Date(request.timestamp as number)) // indexed
+      .andWhere(REQUESTS_COLUMNS.address, request.account)
+      .andWhere(REQUESTS_COLUMNS.hashedPhoneNumber, request.hashedPhoneNumber as string) // TODO(Alec) why is this not always defined?
```

Comment:
This is the caller's phone number identifier, which may not be added if the caller is querying their own pepper for the first time.